src/Images/MapWeightSelector.jsx

```diff
@@ -1,73 +1,52 @@
-import React, { useEffect, useRef } from 'react'
+import React, { useState } from 'react'
 import { useSelector } from '@xstate/react'
 import Form from 'react-bootstrap/Form'
 import '../style.css'
 
 function MapWeightSelector(props) {
   const { service } = props
-  const labelImageWeightUIGroup = useRef(null)
-  const labelSelector = useRef(null)
   const send = service.send
-
   let labelMapWeights = [{ name: 'All', value: 'all' }]
   const name = useSelector(
     service,
     (state) => state.context.images.selectedName
   )
-  const actorContext = useSelector(
-    service,
-    (state) => state.context.images.actorContext
-  )
-  const actorContextName = useSelector(service, (state) =>
-    state.context.images.actorContext.get(state.context.images.selectedName)
-  )
   const labelNames = useSelector(
     service,
     (state) =>
       state.context.images.actorContext.get(state.context.images.selectedName)
         .labelNames
   )
+
   labelNames.forEach((value, key) => {
     labelMapWeights.push({
       name: value,
       value: key
     })
   })
 
-  useEffect(() => {
-    service.machine.context.images.labelImageWeightUIGroup =
-      labelImageWeightUIGroup.current
-    service.machine.context.images.labelSelector = labelSelector.current
-  }, [])
-
-  const currentMapWeight = () => {
-    if (actorContext) {
-      const actorContext = actorContextName
-      return actorContext.selectedLabel
-    }
-    return ''
-  }
+  const [weightValue, setWeightValue] = useState(labelMapWeights[0].name)
 
-  const handleChange = (label) => {
-    console.log(label)
+  const handleChange = (event) => {
+    setWeightValue(event.target.value)
     send({
       type: 'LABEL_IMAGE_SELECTED_LABEL_CHANGED',
-      data: { name, selectedLabel: label }
+      data: { name, selectedLabel: weightValue }
     })
   }
 
   return (
-    <div ref={labelImageWeightUIGroup} className="uiGroup">
+    <div className="uiGroup">
       <div className="uiRow">
         <Form.Control
           as="select"
-          ref={labelSelector}
+          value={weightValue}
           onChange={(e) => {
-            handleChange(e.target.value)
+            handleChange(e)
           }}
         >
           {labelMapWeights.map((weight, idx) => (
-            <option key={idx} value={currentMapWeight()}>
+            <option key={idx} value={weight.value}>
```

Comment:
@PaulHax I've been getting this warning: "A component is changing a controlled input to be uncontrolled. This is likely caused by the value changing from a defined to undefined, which should not happen"
Reading about controlled vs uncontrolled components and bugs that people have had before, I thought doing `value={weight.value || ''}` would fix that issue: 
![image](https://user-images.githubusercontent.com/60415505/174313668-aaf9f29f-74ef-4a51-9551-d5bafa7228b4.png)
but it does not. Might you have thoughts on this? Thank you.

Reply:
I think you are on the right track.  Did you find the case where the value becomes undefined?  Perhaps run some logic to clean up the labelMapWeights array, then map over it.

Reply:
Thanks for the suggestion. I tried multiple approaches here and could not find an undefined variable. Here is a reduced code I created to test things (note for futre self to keep investigating): 
 
 `import React from 'react'
import { useSelector } from '@xstate/react'
import Form from 'react-bootstrap/Form'
import '../style.css'

function MapWeightSelector(props) {
  const { service } = props
  const send = service.send
  let labelMapWeights = [
    { name: 'All', value: 'all' },
    { name: '0', value: '0' },
    { name: '1', value: '1' },
    { name: '2', value: '2' }
  ]
  const name = useSelector(
    service,
    (state) => state.context.images.selectedName
  )

  //tests
  console.log(
    `lanelMapWeights[0].name  typeof=${typeof labelMapWeights[0]
      .name}, lanelMapWeights[0].value  typeof=${typeof labelMapWeights[0]
      .value}`
  )

  console.log(
    `lanelMapWeights[1].name typeof=${typeof labelMapWeights[1]
      ?.name}, lanelMapWeights[1].value typeof=${typeof labelMapWeights[1]
      ?.value}`
  )
  console.log(`labelMapWeights.size = ${Object.keys(labelMapWeights).length}`)
  //end-tests

  const handleChange = (event) => {
    const value = event.target.value
    send({
      type: 'LABEL_IMAGE_SELECTED_LABEL_CHANGED',
      data: { name: name, selectedLabel: value }
    })
  }

  return (
    Object.keys(labelMapWeights).length === 4 && (
      <div className="uiGroup">
        <div className="uiRow">
          <Form.Control
            as="select"
            onChange={(e) => {
              handleChange(e)
            }}
          >
            <option key={`all`} value={`all`}>
              All
            </option>
            <option key={0} value={0}>
              0
            </option>
            <option key={1} value={1}>
              1
            </option>
            <option key={2} value={2}>
              2
            </option>
          </Form.Control>
        </div>
      </div>
    )
  )
}

export default MapWeightSelector
`